### PR TITLE
[MIG][15.0] openupgrade_framework

### PR DESCRIPTION
**Pull request by jcdrubay**
## WIP

- ...

## Unresolved Issues

- https://github.com/jcdrubay/OpenUpgrade/issues

## Done

- pre-migration: Add call to rename/merge modules listed in apriori.py
- Rename the xml ids which have been moved from the module mail to the module base
- Remove the odoo_patch on function _postprocess_view which became useless after the input argument `validate` has been removed from native Odoo

### CODE REVIEW

**Comment by legalsylvain** [comment]
First commit is not desired i think. Otherwise. Lgtm. Thanks for the contribution !

**Comment by jcdrubay** [comment]
Without the first commit, the upgrade cannot run... anyway, we can handle in 2 separate PRs.

I will go "all the way" with this PR including removing the obsolete function.
I will consider this PR as a migration PR.

I prefer that option, so I can continue pushing code with separate commits to handle whichever issue I may bump into.

Creating one PR for one fix will be very tedious + I am not sure if the fix is good until I can test when there will be no more issues...

I hope this works for you... later on, we can re-create one PR per issue if you prefer.

I am updating the PR summary.

**Comment by jcdrubay** [comment]
@legalsylvain FYI, I got a first successful run for the upgrade my project which depends on very few native odoo modules today. 

Then, I will be:

- reviewing the analysis file of each module to check if I missed something.
- testing more

I will create dedicated PR for each module when I see fit. Should I continue submitting PR to your repo or should I switch to OCA repo?
I started with this one: https://github.com/OCA/OpenUpgrade/pull/3242

If you agree we could merge this PR? 
I could squash the commits if you want. Let me know.

Also, it would be great  if you could give me some inputs for the 2 non-blocking opened issues I have.
https://github.com/jcdrubay/OpenUpgrade/issues

**Comment by jcdrubay** [comment]
I think that all that need to be done for base is done.

File: openupgrade_framework/odoo_patch/odoo/addons/base/models/ir_ui_view.py [hunk 1]
```diff
@@ -45,17 +45,7 @@ def _raise_view_error(
             )
             self.write({"active": False})
 
-
-def _postprocess_view(self, node, model, validate=True, editable=True):
-    """Don't validate views, _raise_view_error is mutted"""
-    return View._postprocess_view._original_method(
-        self, node, model, validate=False, editable=editable
-    )
```
Comment (legalsylvain) [inline openupgrade_framework/odoo_patch/odoo/addons/base/models/ir_ui_view.py:53]: naive question : there is no blocking error if views are broken ?

File: openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py [hunk 2]
```diff
@@ -34,4 +34,40 @@ def migrate(cr, version):
             "recommended to run the Odoo with --load=openupgrade_framework "
             "when migrating your database."
         )
-        openupgrade.rename_xmlids(cr, [("mail.icp_mail_catchall_alias", "base.icp_mail_catchall_alias")])
+
+    openupgrade.rename_xmlids(
+        cr,
+        [
+            # From l10n_ec
+            ("l10n_ec.state_ec_1", "base.state_ec_01"),
```
Comment (legalsylvain) [inline openupgrade_scripts/scripts/base/15.0.1.3/pre-migration.py]: nitpicking (non blocking) : usually, in Openupgrade, we define a variable at the top of the file, and then just call ``rename_xml_ids`` in the main function. It is so more readable to see all the steps of the main file.

see for exemple : https://github.com/OCA/OpenUpgrade/blob/14.0/openupgrade_scripts/scripts/base/14.0.1.3/pre-migrate.py#L23

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Thanks @jcdrubay for your work ! 

> Creating one PR for one fix will be very tedious + I am not sure if the fix is good until I can test when there will be no more issues...

:100: 

Indeed. This PR is the perfect way to move one.

> I think that all that need to be done for base is done.

Nice news ! 

Review : 
- code review, LGTM.
- Test review : it is failing. Reason : ``test.yml`` is outdated.  the test try to restore a V13 database, and then realize the migration script 14->15. Quite normal it is failing ;-) could you change that file ? https://github.com/legalsylvain/OpenUpgrade/blob/15.0-mig-openupgrade_framework/.github/workflows/test.yml#L47
(some 13 should be renamed in 14 and some 14 in 15...)

**Comment by jcdrubay** [comment]
@legalsylvain Is it ok now?

- [1] in the .github/workflows/test.yml: I updated versions (not 100% sure if it's right, to test, I created this PR: https://github.com/OCA/OpenUpgrade/pull/3248)
- [2] I used a variable rename_xmlids which is then called by the migrate function
- [3] I answered your question related to the _postprocess_view function and removal of the validate argument

😄  🤞

**Comment by legalsylvain** [comment]
Test are green ! Nice. Some bug in precommit install. Otherwise. Lgtm !

**Comment by etobella** [comment]
@legalsylvain I think https://github.com/jcdrubay/OpenUpgrade/pull/5 can solve the pre-commit issue

**Comment by etobella** [comment]
I thin ocabot will not work on your repo :thinking:

**Comment by legalsylvain** [comment]
Let's try again ?

/ocabot merge nobump

**Comment by legalsylvain** [comment]
weird

**Pull request merged**